### PR TITLE
Initial code for for AVM - Azure Bastion Module

### PR DESCRIPTION
**Pull request by herms14**
- Azure Bastion module for AVM

### CODE REVIEW

**Comment by herms14** [comment]
For pending review by @Azure/avm-core-team

File: variables.tf
Comment (mbilalamjad) [inline variables.tf:183]: Tags variable doesn't have a description needed for documentation

File: variables.tf
Comment (mbilalamjad) [inline variables.tf:68]: Providing a default value for this will make it optional, recommended instead to provide a detailed description which is missing for this variable.

File: terraform.tf
Comment (mbilalamjad) [inline terraform.tf:6]: Please follow [Spec TFNFR26](https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tfnfr26---category-code-style---provider-version-constraint-must-have-a-constraint-on-maximum-major-version)

File: variables.tf
Comment (mbilalamjad) [inline variables.tf:61]: There's a tags variable being declared separately also which should be used instead of this one.

File: variables.tf
Comment (mbilalamjad) [inline variables.tf:1]: Maybe I missed it but haven't seen virtual_network_name & subnet_name variables being used

File: examples/default/outputs.tf
Comment (mbilalamjad) [inline examples/default/outputs.tf:1]: Seems like bastion_host_name and bastion_host_id are the same as bastion_host

File: _footer.md
Comment (mbilalamjad) [inline _footer.md:9]: Recommended to move this notice into the header

File: SUPPORT.md
Comment (mbilalamjad) [inline SUPPORT.md:1]: Recommended to update this file

**Review by mbilalamjad — CHANGES_REQUESTED** [review]
Hi @herms14, thank you for requesting a review of the PR and module, have provided my feedback, appreciate if you could look into it and in case you have any questions or concerns than please feel free to reach out to me.

**Comment by herms14** [comment]
Hi @mbilalamjad ,

Thanks for the review. I have updated the code to reflect your feedback.

**Review by mbilalamjad — APPROVED** [review]
Thank you for the updates

**Comment by mbilalamjad** [comment]
@herms14 recommend looking into the tests failing before merging and publishing version 0.1.0 module to the terraform registry.

**Comment by herms14** [comment]
will do @mbilalamjad  thanks for the review

**Pull request merged**
